xapian-letor/ranker/ranker.cc

```diff
@@ -192,32 +193,31 @@ write_to_file(const std::vector<Xapian::FeatureVector> & list_fvecs, const strin
     }
 }
 
+// Query file is in the format: <qid> '<query_string>'
+// Although it will accept any number of arbitrary characters between
+// the first space and the first single quote.
 static std::pair<string, string>
 parse_query_string(const string & query_line, int line_number)
 {
-    vector<string> token;
-    size_t j = 0;
-    while (j < query_line.size()) {
-	size_t i = query_line.find_first_not_of(' ', j);
-	if (i == string::npos) break;
-	j = query_line.find_first_of(' ', i);
-	token.push_back(query_line.substr(i, j - i));
+    Assert(!query_line.empty());
+    string::size_type j = query_line.find_first_of(' ');
```

Comment:
Also lets make sure that its mentioned in the docs that space is used as a delimiter between query and query id. Right now its quite vague as this is the description ( although the delimiter looks like a space, it would be better to mention it ).

> Query` file: This file has information of queries to be involved in learning and its id. It should be formatted in such a way:

> 2010001 'landslide,malaysia'
> 2010002 'search,engine'
> 2010003 'Monuments,of,India'
> 2010004 'Indian,food'

> where 2010xxx being query-id followed by a comma separated query in single-quotes.

Reply:
I'd agree it would be better to be more precise about what separators are allowed between the fields.

Are commas separating terms required?  The docs currently give examples such as `2010003 'Monuments of India'`, but then just below that say "a comma separated query".  The code seems to parse these strings using `Xapian::QueryParser::parse_query()` which will work with spaces or commas between terms.

If this is an externally-specified file format which says commas are required, then we should probably follow that.  But if this is a xapian-letor-specific format, it seems more helpful to allow query strings as entered by the user.

Looking at what the docs currently say, I also noticed "All the fields are whitespace delimited" about the Qrel file format, which appears to be incorrect - the parsing in `load_relevance()` in `xapian-letor/ranker/ranker.cc` seems to look for spaces, whereas even if "whitespace" is read as "horizontal whitespace" that includes tabs.  Requiring spaces seems OK (unless there's an external specification for this format which says tabs are allowed) - the problem here is the docs not matching the code.

Reply:
> If this is an externally-specified file format which says commas are required, then we should probably follow that. But if this is a xapian-letor-specific format, it seems more helpful to allow query strings as entered by the user.
>
I think space separated query are fine as the same format was used in FIRE Dataset.
I have updated the docs accordingly.